### DISCUSSION TITLE
Error while running Cypress 9.1.0 on node 18.12.0

### DISCUSSION THREAD

**Discussion by phillmund** [cypress-io/cypress#24544] [Questions and Help]
After upgrading to the new node lts version I get the following error while running the existing tests via the open command:

`Error: The following error was thrown by a plugin. We stopped running your tests because a plugin crashed. Please check your plugins file (/***/cypress/plugins/index.js)
    at Object.get (/***/Library/Caches/Cypress/9.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/errors.js:1043:15)
    at EventEmitter.handleError (/***/Library/Caches/Cypress/9.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/index.js:189:20)
    at EventEmitter.emit (node:events:394:28)
    at ChildProcess.<anonymous> (/***/Library/Caches/Cypress/9.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:19:22)
    at ChildProcess.emit (node:events:394:28)
    at emit (node:internal/child_process:920:12)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)`

My plugins file is empty:

`/// <reference types="cypress" />
// ***********************************************************
// This example plugins/index.js can be used to load plugins
//
// You can change the location of this file or turn off loading
// the plugins file with the 'pluginsFile' configuration option.
//
// You can read more here:
// https://on.cypress.io/plugins-guide
// ***********************************************************

// This function is called when a project is opened or re-opened (e.g. due to
// the project's config changing)

/**
 * @type {Cypress.PluginConfig}
 */
// eslint-disable-next-line no-unused-vars
module.exports = (on, config) => {
  // "on" is used to hook into various events Cypress emits
  // "config" is the resolved Cypress config
}
`

Did anyone have this issue before? How can I solve it? Thanks in advance!

**Comment by portyu9** [comment]
Hi @phillmund 

I would not run Cypress `9.1.0` on Node `18.12.0`.

Cypress 9.1 predates Node 18, and the plugin/config process runs in Node, which explains why you can get a "plugins file crashed" error even when `plugins/index.js` is empty.

For the historical setup, the least disruptive fix is to pin the project to the Node version supported by that Cypress generation (for example Node 16) and retry.

The better long-term fix is to upgrade Cypress, but don't jump several major versions without reviewing the migration guides because Cypress 10+ changed configuration/component testing substantially, Cypress 12 introduced test-isolation changes, and later releases have further breaking changes.

Also note that **current** Cypress has moved on again and no longer supports Node 18, so simply installing today's Cypress while remaining on Node 18 is not the migration target either.

In short: keep the Cypress and Node major versions compatible rather than trying to fix this in the empty plugins file.

Please mark this answer as accepted if it helped, thank you.
